Problem with extra coordinate on 32-bit system

While running tests on a 32-bit system, I got the following errors (this is with version 1.0.1, but I obtain similar results with the master version):
```
============================= test session starts ==============================
platform linux -- Python 3.7.1, pytest-3.6.4, py-1.7.0, pluggy-0.8.0
rootdir: /build/1st/ndcube-1.0.1, inifile: setup.cfg
plugins: remotedata-0.3.1, openfiles-0.3.0, doctestplus-0.1.3, arraydiff-0.2
collected 95 items / 4 errors

==================================== ERRORS ====================================
_________________ ERROR collecting ndcube/tests/test_ndcube.py _________________
../../../ndcube/tests/test_ndcube.py:123: in <module>
    [(cubem[:, 1],
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
------------------------------- Captured stderr --------------------------------
WARNING: FITSFixedWarning: 'celfix' made the change 'Unmatched celestial axes'. [astropy.wcs.wcs]
/build/1st/ndcube-1.0.1/ndcube/tests/test_ndcube.py:65: RuntimeWarning: invalid value encountered in sqrt
  uncertainty = np.sqrt(data)
_________________ ERROR collecting ndcube/tests/test_ndcube.py _________________
../../../ndcube/tests/test_ndcube.py:123: in <module>
    [(cubem[:, 1],
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
INFO: uncertainty should have attribute uncertainty_type. [astropy.nddata.nddata]
------------------------------- Captured stderr --------------------------------
/build/1st/ndcube-1.0.1/ndcube/tests/test_ndcube.py:65: RuntimeWarning: invalid value encountered in sqrt
  uncertainty = np.sqrt(data)
_____________ ERROR collecting ndcube/tests/test_ndcubesequence.py _____________
../../../ndcube/tests/test_ndcubesequence.py:115: in <module>
    (seq[0], NDCube),
../../../ndcube/ndcube_sequence.py:84: in __getitem__
    return utils.sequence.slice_sequence(self, item)
../../../ndcube/utils/sequence.py:57: in slice_sequence
    return slice_sequence_by_sequence_items(cubesequence, sequence_items)
../../../ndcube/utils/sequence.py:211: in slice_sequence_by_sequence_items
    return result.data[sequence_items[0].sequence_index][sequence_items[0].cube_item]
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
_____________ ERROR collecting ndcube/tests/test_ndcubesequence.py _____________
../../../ndcube/tests/test_ndcubesequence.py:115: in <module>
    (seq[0], NDCube),
../../../ndcube/ndcube_sequence.py:84: in __getitem__
    return utils.sequence.slice_sequence(self, item)
../../../ndcube/utils/sequence.py:57: in slice_sequence
    return slice_sequence_by_sequence_items(cubesequence, sequence_items)
../../../ndcube/utils/sequence.py:211: in slice_sequence_by_sequence_items
    return result.data[sequence_items[0].sequence_index][sequence_items[0].cube_item]
../../../ndcube/mixins/ndslicing.py:31: in __getitem__
    return super().__getitem__(item)
/usr/lib/python3/dist-packages/astropy/nddata/mixins/ndslicing.py:61: in __getitem__
    return self.__class__(**kwargs)
../../../ndcube/ndcube.py:185: in __init__
    extra_coords, self.missing_axis, data.shape)
../../../ndcube/utils/cube.py:90: in _format_input_extra_coords_to_extra_coords_wcs_axis
    raise ValueError(coord_1_format_error)
E   ValueError: 2nd element of extra coordinate tuple must be None or an int giving the data axis to which the coordinate corresponds.
------------------------------- Captured stdout --------------------------------
[{'coordinate_type': None, 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'spectral', 'scale': 'linear', 'group': 0, 'number': 0}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 1}, {'coordinate_type': 'celestial', 'scale': 'non-linear celestial', 'group': 0, 'number': 0}]
!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!!
```

--- a/changelog/269.bugfix.rst
+++ b/changelog/269.bugfix.rst
@@ -1,0 +1,1 @@
+Generalize int type checking so it is independent of the bit-type of the OS.

--- a/ndcube/mixins/sequence_plotting.py
+++ b/ndcube/mixins/sequence_plotting.py
@@ -1,4 +1,5 @@
 import copy
+import numbers
 
 import astropy.units as u
 import matplotlib as mpl
@@ -1036,7 +1037,7 @@ class LineAnimatorNDCubeSequence(LineAnimator):
                         raise TypeError("Extra coord {} must be of same type for all NDCubes to "
                                         "use it to define a plot axis.".format(axis_extra_coord))
                     if extra_coord_axes.all() == extra_coord_axes[0]:
-                        if isinstance(extra_coord_axes[0], (int, np.int64)):
+                        if isinstance(extra_coord_axes[0], numbers.Integral):
                             extra_coord_axes = [int(extra_coord_axes[0])]
                         else:
                             extra_coord_axes = list(extra_coord_axes[0]).sort()

--- a/ndcube/utils/cube.py
+++ b/ndcube/utils/cube.py
@@ -2,6 +2,7 @@
 """
 Utilities for ndcube.
 """
+import numbers
 
 import numpy as np
 from astropy.units import Quantity
@@ -46,7 +47,7 @@ def data_axis_to_wcs_ape14(data_axis, pixel_keep, naxes, old_order=False):
 
     # Make sure that data_axis is a scalar item
     if data_axis is not None:
-        if not isinstance(data_axis, (int, np.int32, np.int64)):
+        if not isinstance(data_axis, numbers.Integral):
             raise ValueError(f"The data_axis parameter accepts \
                 numpy.int64 or numpy.np.int32 datatype, got this {type(data_axis)}")
 
@@ -124,7 +125,7 @@ def wcs_axis_to_data_ape14(wcs_axis, pixel_keep, naxes, old_order=False):
 
     # Make sure that wcs_axis is a scalar item
     if wcs_axis is not None:
-        if not isinstance(wcs_axis, (int, np.int32, np.int64)):
+        if not isinstance(wcs_axis, numbers.Integral):
             raise ValueError(f"The wcs_axis parameter accepts \
                 numpy.int64 or np.int32 datatype, got this {type(wcs_axis)}")
 
@@ -218,8 +219,7 @@ def _format_input_extra_coords_to_extra_coords_wcs_axis(extra_coords, pixel_keep
             raise ValueError(coord_format_error.format(coord))
         if not isinstance(coord[0], str):
             raise ValueError(coord_0_format_error.format(coord))
-        if coord[1] is not None and not isinstance(coord[1], int) and \
-                not isinstance(coord[1], np.int64):
+        if coord[1] is not None and not isinstance(coord[1], numbers.Integral):
             raise ValueError(coord_1_format_error)
 
         # Determine wcs axis corresponding to data axis of coord

